xDSCResourceDesigner.psm1

```diff
@@ -2242,152 +2334,162 @@ function Test-MockSchema
         $errorIdsRef
     )
 
-    # Returns full path to a 0 byte .tmp file
-    $tempFilePath = [IO.Path]::GetTempFileName() 
+    $newSchemaPath = $null
+    $newSchemaName = $null
 
-    $tempFolderPath = [IO.Path]::GetDirectoryName($tempFilePath)
-
-    # Extracts the ????.tmp name
-    $newSchemaName = [IO.Path]::GetFileNameWithoutExtension($tempFilePath)
+    try
+    {
+        # Returns full path to a 0 byte .tmp file
+        $tempFilePath = [IO.Path]::GetTempFileName() 
 
-    # We can now use the temp file name to create a new unique file
-    Remove-Item $tempFilePath
+        $tempFolderPath = [IO.Path]::GetDirectoryName($tempFilePath)
 
-    $newSchemaPath = "$tempFolderPath\$newSchemaName.schema.mof"
+        # Extracts the ????.tmp name
+        $newSchemaName = [IO.Path]::GetFileNameWithoutExtension($tempFilePath)
 
-    $null = New-Item $newSchemaPath -ItemType File
+        # We can now use the temp file name to create a new unique file
+        Remove-Item $tempFilePath
 
-    $null = [IO.Path]::GetFileNameWithoutExtension($Schema) -cmatch "(.+)\.schema"
-    $schemaName = $Matches[1]
+        $newSchemaPath = "$tempFolderPath\$newSchemaName.schema.mof"
 
-    # Initialize this to correct; it will be overwritten if incorrect
-    $className = $schemaName
+        $null = New-Item $newSchemaPath -ItemType File
 
-    $extendsOMI = $false
+        $null = [IO.Path]::GetFileNameWithoutExtension($Schema) -cmatch "(.+)\.schema"
```

Comment:
To clarify, if you look at the original file, there's a bunch of code, followed by this, which is pointless:

```posh
    try{}
    catch{}
    finally{
       Remove-WmiObject -Class $newSchemaName -Namespace root\microsoft\windows\DesiredStateConfiguration -ErrorAction SilentlyContinue
       Remove-Item $newSchemaPath -ErrorAction SilentlyContinue
    }   
```

That block happens after a `return` statement earlier in the code, and so is never executed.  What I did was move all of the preceding code into a `try` block, indented it, removed the empty `catch` block, and used the same `finally` block.